### PR TITLE
[project] added rename button on panel

### PR DESCRIPTION
**Pull request by smallfont00**
should done #154

### CODE REVIEW

File: project/panel.rkt [hunk 1]
```diff
@@ -108,7 +108,8 @@ modifier author: Lîm Tsú-thuàn(GitHub: @dannypsnl)
               (let loop ()
                 (match (file-watcher-channel-get)
                   [(or (list 'robust 'add _)
-                       (list 'robust 'remove _))
+                       (list 'robust 'remove _)
+                       (list 'robust 'rename _))
```
Comment (dannypsnl) [inline project/panel.rkt]: 你確定有這個 event？

File: project/panel.rkt [hunk 2]
```diff
@@ -150,13 +151,27 @@ modifier author: Lîm Tsú-thuàn(GitHub: @dannypsnl)
     (define (remove-path-and-refresh btn event)
       (delete-directory/files (send view get-cur-selected-file) #:must-exist? #f)
       (send view reset-directory (send current-project get)))
+    (define (rename-path-and-refresh btn event)
+      (define selected-dir (send view get-cur-selected-dir))
+      (define selected-file (send view get-cur-selected-file))
+      (define name (get-text-from-user "new name of selected path?" ""))
```
Comment (dannypsnl) [inline project/panel.rkt]: new name for

File: project/panel.rkt [hunk 3]
```diff
@@ -150,13 +151,27 @@ modifier author: Lîm Tsú-thuàn(GitHub: @dannypsnl)
...
+      (define old-path (or selected-file selected-dir))
+      (define new-path (if (equal? selected-dir selected-file)
+                           (build-path selected-dir 'up name)
```
Comment (dannypsnl) [inline project/panel.rkt]: 嗯嗯 其實我沒想這麼複雜，我的想法是從當前專案路徑為基礎建立相對路徑
你要的話也可以改成該檔案到當前目錄為基礎

**Review by dannypsnl — CHANGES_REQUESTED** [review]

File: project/panel.rkt (hunk 1 above)
Comment (dannypsnl) [reply project/panel.rkt]: 你可以看 file-watchers 這個 collection 的文件

Comment (smallfont00) [reply project/panel.rkt]: 我並不知道這是幹嘛的，我看add和remove都有我就把rename加上去了😂

File: project/panel.rkt (hunk 3 above)
Comment (smallfont00) [reply project/panel.rkt]: 這能只用get-cur-selected-dir和get-cur-selected-file寫出來嗎?.....

File: project/panel.rkt (hunk 1 above)
Comment (dannypsnl) [reply project/panel.rkt]: https://docs.racket-lang.org/file-watchers/index.html?q=file-watchers#%28def._%28%28lib._file-watchers%2Fmain..rkt%29._robust-watch%29%29

this is created to detect external file changes

File: project/panel.rkt (hunk 3 above)
Comment (dannypsnl) [reply project/panel.rkt]: Yes, you can check `make-parent-directory*`

File: project/panel.rkt (hunk 2 above)
Comment (dannypsnl) [reply project/panel.rkt]: by the way, give it init-val and #f parent. ref to https://docs.racket-lang.org/gui/Windowing_Functions.html?q=get-text-from-user#%28def._%28%28lib._mred%2Fmain..rkt%29._get-text-from-user%29%29

File: project/panel.rkt [hunk 4]
```diff
@@ -150,13 +153,26 @@ modifier author: Lîm Tsú-thuàn(GitHub: @dannypsnl)
     (define (remove-path-and-refresh btn event)
       (delete-directory/files (send view get-cur-selected-file) #:must-exist? #f)
       (send view reset-directory (send current-project get)))
+    (define (rename-path-and-refresh btn event)
+      (define name (get-text-from-user "new name for selected path?" ""))
```
Comment (dannypsnl) [inline project/panel.rkt]: still missing `parent` and `init-val`

**Review by dannypsnl — CHANGES_REQUESTED** [review]

File: project/panel.rkt (hunk 4 above)
Comment (dannypsnl) [reply project/panel.rkt]: 1. `init-val` here: origin path
2. `parent` here: `#f`

Comment (smallfont00) [reply project/panel.rkt]: I am sorry. I didn't see all the message from the last requested changes.

Comment (dannypsnl) [reply project/panel.rkt]: No worry, this is why we do reviewing

**Review by dannypsnl — APPROVED** [review]

**Pull request merged**